src/main/java/io/codiga/plugins/jetbrains/model/CodingAssistantCodigaTransform.java

```diff
@@ -0,0 +1,95 @@
+package io.codiga.plugins.jetbrains.model;
+
+import com.intellij.ide.macro.Macro;
+import com.intellij.ide.macro.MacroManager;
+import io.codiga.plugins.jetbrains.assistant.transformers.*;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class CodingAssistantCodigaTransform {
+  private final CodingAssistantContext CodigaTransformationContext;
+
+  public CodingAssistantCodigaTransform(CodingAssistantContext CodigaTransformationContext) {
+    this.CodigaTransformationContext = CodigaTransformationContext;
+  }
+
+  /**
+   * Search for Codiga's recipe variables in raw code string and transform
+   * any if found into expected value or behavior. If a new variable is
+   * supported, include it in the VARIABLE_TO_TRANSFORMER HashMap and create
+   * a new class for it with its `transform` method inside
+   * assistant.transformers.
+   *
+   * @param code
+   * @return code string with Codiga's recipe variables transform to local if
+   * any was found.
+   */
+  public String findAndTransformVariables (String code) {
+    Map<String, VariableTransformer> VARIABLE_TO_TRANSFORMER = new HashMap();
```

Comment:
This Map is immutable. We should build it in the constructor and not every time we do a transformation. Move the `Map` as a class member and build it in the constructor.